src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/System.Text.RegularExpressions.Tests.csproj

```diff
@@ -61,6 +63,9 @@
     <Compile Include="RegexExperiment.cs" />
     <Compile Include="RegexGeneratorHelper.netcoreapp.cs" />
     <Compile Include="$(CommonTestPath)System\Diagnostics\DebuggerAttributes.cs" Link="Common\System\Diagnostics\DebuggerAttributes.cs" />
+
+    <PackageReference Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="$(MicrosoftCodeAnalysisVersion)" PrivateAssets="all" />
+    <PackageReference Include="Microsoft.CodeAnalysis.CSharp.CodeFix.Testing.XUnit" Version="$(CompilerPlatformTestingVersion)" />
```

Comment:
What are these references and the CSharpCodeFixVerifier`2.cs needed for?

Reply:
Oh, I see, the existing support we have for validating the analyzer isn't sufficient for actually making changes, hence needing the additional helpers and workspace support.

Reply:
Apart from needing to validate the code fixer as  you point out, this verifier also adds the possibility to validate the exact location of where the diagnostic was emitted, which I believe wasn't possible with the existing source generator infra.